.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,15 @@
+on:
+  pull_request:
+
+jobs:
+  pre-commit:
+    name: pre-commit linting
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v4
+      with:
+        python-version: 3.x
+    - uses: pre-commit/action@v3.0.0
+    - uses: pre-commit-ci/lite-action@v1.0.1
+      if: always()
```

Comment:
This means, that also in case of a non zero exit of a previous step the action will be performed. This action triggers the actual commit of the changes made by pre-commit, so we do want to run this step in exactly the case of a non-zero exit before.